server/middleware/clients.coffee

```diff
@@ -0,0 +1,16 @@
+errors = require '../commons/errors'
+wrap = require 'co-express'
+database = require '../commons/database'
+Client = require '../models/Client'
+
+newSecret = wrap (req, res, next) ->
+  client = yield database.getDocFromHandle(req, Client)
+  if not client
+    throw new errors.NotFound('Client not found.')
+  secret = client.generateNewSecret()
+  yield client.save()
```

Comment:
This generate/save sequence is a little confusing unless you look at generate impl.  What if we called it client.addNewSecret or client.setNewSecret instead?